### ISSUE TITLE
Division by zero in Overhang

### ISSUE THREAD

**Issue by damienpicard** [open-ideas/IDEAS#397]
A division by zero occurs in `IDEAS.Buildings.Components.Shading.Overhang` when `dep=0`.
This is clear from these equations:

```
y1*Modelica.Math.cos(verAzi) = dep*Modelica.Math.tan(alt);
x1 = dep*Modelica.Math.tan(verAzi);
shdwTrnglRtio*x1 = y1;
``` 

However, I do not understand the model very well and I'm not sure how to fix it.
@Mathadon @GlennReynders @cprotopa @rubenbaetens , do you know this model?

We should fix this as it causes problem for overhang with vertical panes and no horizontal pane.

**Comment by cprotopa** [comment]
I didn't make these, so I'm not so sure, but wouldn't you use the `SideFins` model in this case? And if you have both, the `Box` model. Now of course if you want to vary the depth down to 0, I suppose you'd have to fix this. I think @Mathadon made the models..?

**Comment by damienpicard** [comment]
@Mathadon did you make this model? Or is it from @rubenbaetens ?

**Comment by rubenbaetens** [comment]
I made the model, it's a copy from the model as it is implemented in EnergyPlus if i remember well.

**Comment by damienpicard** [comment]
dep=0 is not a physical system. Use then either side vertical pane, horizontal pane or box pane.
@GlennReynders  add an assert() to get an error message for this non-physical configuration.

**Comment by rubenbaetens** [comment]
Addressed in #433.
